Services/Authentication/classes/class.ilAuthModeDetermination.php

```diff
@@ -113,11 +113,26 @@ public function getAuthModeSequence(string $a_username = ''): array
                     //#17731
                     $pattern = str_replace('*', '.*?', $server->getUsernameFilter());
 
-                    if (preg_match('/^' . $pattern . '$/', $a_username)) {
-                        $this->logger->debug('Filter matches for ' . $a_username);
-                        array_unshift($sorted, $auth_key);
-                        continue;
+                    foreach (['/', '~', '@', ';', '%', '`', '#'] as $delimiter) {
+                        $pattern_string = $delimiter . "^" . $pattern . '$' . $delimiter . 'i';
+                        $this->logger->debug(': Match pattern:' . $pattern . ' => ' . $a_username);
+                        set_error_handler(static function (int $severity, string $message, string $file, int $line): never {
+                            throw new ErrorException($message, $severity, $severity, $file, $line);
+                        });
+
+                        try {
+                            if (preg_match($pattern_string, $a_username) === 1) {
+                                $this->logger->debug('Filter matches for ' . $a_username);
+                                array_unshift($sorted, $auth_key);
+                                continue 2;
+                            }
```

Comment:
If the RegEx evaluation does not throw an exception, we should continue as well.